### PR TITLE
Adopt `:autosummary:` in `torchaudio.transforms` module doc

### PR DESCRIPTION
**Pull request by mthrok**
* Introduce the mini-index at `torchaudio.transforms` page.
* Add "Augmentations" subsection. 
* Also updated the overall introduction.

https://output.circle-artifacts.com/output/job/1b65246a-403c-4d2c-b97d-d1b582d8b4e5/artifacts/0/docs/transforms.html

Screen Shot 2022-09-16 at 5 20 08 PM

Screen Shot 2022-09-16 at 5 20 28 PM

### CODE REVIEW

File: docs/source/transforms.rst [hunk 1]
```diff
@@ -1,219 +1,144 @@
...
+   # Move the computation graph to CUDA
+   pipeline.to(device=torch.device("CUDA"))
 
-:hidden:`ComputeDeltas`
------------------------
+   # Perform the transform
+   features = pipeline(waveform)
 
-.. autoclass:: ComputeDeltas
+Please checkout tutorials that cover in-depth usage of trasforms.
```
Comment (carolineechen) [inline docs/source/transforms.rst:72]:
```suggestion
Please check out tutorials that cover in-depth usage of trasforms.
```

File: docs/source/transforms.rst [hunk 2]
```diff
@@ -1,219 +1,144 @@
...
-:hidden:`TimeMasking`
----------------------
-
-.. autoclass:: TimeMasking
-
-  .. automethod:: forward
-
-:hidden:`TimeStretch`
----------------------
+:mod:`torchaudio.transforms` module contains common audio processings and feature extractions. The following diagram shows the relationship between some of the transforms available.
```
Comment (carolineechen) [inline docs/source/transforms.rst:6]:
```suggestion
:mod:`torchaudio.transforms` module contains common audio processings and feature extractions. The following diagram shows the relationship between some of the available transforms.
```

File: docs/source/transforms.rst [hunk 3]
```diff
@@ -1,219 +1,144 @@
...
 
-.. autoclass:: TimeStretch
 
-  .. automethod:: forward
+.. image:: https://download.pytorch.org/torchaudio/tutorial-assets/torchaudio_feature_extractions.png
 
-:hidden:`Fade`
---------------
+Transforms are implemented using :class:`torch.nn.Module`. Common ways to build a processing pipeline is to define custom Module class or chain Modules together using :class:`torch.nn.Sequential`, then moving it to target device and data type.
```
Comment (carolineechen) [inline docs/source/transforms.rst:11]:
```suggestion
Transforms are implemented using :class:`torch.nn.Module`. Common ways to build a processing pipeline are to define custom Module class or chain Modules together using :class:`torch.nn.Sequential`, then move it to a target device and data type.
```

File: docs/source/transforms.rst [hunk 4]
```diff
@@ -1,219 +1,144 @@
...
-:hidden:`LFCC`
---------------
+.. code::
 
-.. autoclass:: LFCC
+   # Instantiate a pipeline
+   pipeline = MyPipeline()
 
-  .. automethod:: forward
+   # Move the computation graph to CUDA
```
Comment (carolineechen) [inline docs/source/transforms.rst:66]: could be good to move this to a dtype as well for demonstration

File: docs/source/transforms.rst [hunk 5]
```diff
@@ -1,219 +1,144 @@
...
+    PitchShift
+    SlidingWindowCmn
+    SpectralCentroid
+    Vad
+
+Augmentations
 -------------
 
-.. autoclass:: Vad
+The following transforms implements popular augmentation techniques known as *SpecAugment* :cite:`specaugment`.
```
Comment (carolineechen) [inline docs/source/transforms.rst:115]:
```suggestion
The following transforms implement popular augmentation techniques known as *SpecAugment* :cite:`specaugment`.
```

**Review by carolineechen — APPROVED** [review]

File: docs/source/transforms.rst (hunk 4 above)
Comment (mthrok) [reply docs/source/transforms.rst:66]: done

**Pull request closed without merging**